sktime/utils/_testing/_conditional_fixtures.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+"""Testing utility for easy generation of conditional fixtures in pytest_generate_tests.
+
+Exports conditional_fixtures_and_names utility
+"""
+
+__author__ = ["fkiraly"]
+
+__all__ = ["conditional_fixtures_and_names"]
+
+from copy import deepcopy
+
+import numpy as np
+
+
+class FixtureGenerationError(Exception):
+    """Raised when a fixture fails to generate."""
+
+    def __init__(self, fixture_name=""):
+        self.fixture_name = fixture_name
+        fixture_name = fixture_name + " "
+        super().__init__(f"fixture {fixture_name}failed to generate")
+
+
+def conditional_fixtures_and_names(
```

Comment:
```suggestion
def create_conditional_fixtures(
```

Reply:
agreed, this is a better name. Replaced throughout.